source/common/http/conn_manager_impl.cc

```diff
@@ -488,6 +493,15 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
     is_head_request_ = true;
   }
 
+  // Drop new requests when overloaded as soon as we have decoded the headers.
+  if (connection_manager_.overload_stop_accepting_requests_ ==
```

Comment:
You should be able to push this up to the top or after second line of this method.

Reply:
I put it in the current location so it can use the request_headers_ and is_head_request_ fields, similar to all the other sendLocalReply calls in this method. Can some of that be replaced with  hardcoded values instead?

Reply:
Oh, I see, needs `is_head_request_`.. fair enough.